src/plugins/intel_cpu/src/nodes/mvn.cpp

```diff
@@ -815,179 +1281,755 @@ struct jit_uni_mvn_kernel_f32 : public jit_uni_mvn_kernel, public jit_generator
+                    cmp(reg_rt_shape, 0);
+                    jbe(label_end);
+                }
+
+                Xbyak::Label label_sse_full_size;
+                if (isa == cpu::x64::sse41) {
+                    // on sse, first 4 could be done with vector manner
+                    cmp(reg_rt_shape, 4);
+                    jae(label_sse_full_size, T_NEAR);
+                }
+
+                worker_tails_unroll();
+                jmp(label_end, T_NEAR);
+
+                L(label_sse_full_size);
+                {
+                    worker_mvn_vector_unroll(reg_work_amount);
+                    sub(reg_rt_shape, 4);
+                }
+            }
+            L(label_end);
+        }
+    }
+
     // nspc norm per channel with unroll
     inline void norm_nspc_pc_ker() {
-        // 4 unroll vector
-        size_t unroll_size = 4;
-        size_t vec_num = div_up(jcp_.C, vector_step);
-        unroll_size = vec_num >= unroll_size ? unroll_size : vec_num;
-        size_t unroll_number = div_up(vec_num, unroll_size);
+        // stack used as no more GPR.
+        const int gpr_size = 8;
+        sub(rsp, 7 * gpr_size);
+        const Xbyak::Address addr_unroll_size = qword[rsp];
+        const Xbyak::Address addr_unroll_num = qword[rsp + 8];
+        const Xbyak::Address addr_vector_num = qword[rsp + 16];
+        const Xbyak::Address addr_tail_num = qword[rsp + 24];
+        const Xbyak::Address addr_last_unroll_size = qword[rsp + 32];
+        const Xbyak::Address addr_work_amount_bk = qword[rsp + 40];
+        const Xbyak::Address addr_oc_off_bk = qword[rsp + 48];
+
+        // size_t vec_num = C / vector_step
+        mov(rax, reg_rt_shape);
+        mov(addr_vector_num, vector_step);
+        xor_(rdx, rdx);
+        div(addr_vector_num);    // reg_rt_shape / vector_step, rax is result, rdx is tails
+        mov(addr_vector_num, rax);
+        mov(addr_tail_num, rdx);
+
+        // should before tail jmp
+        Xbyak::Reg64 reg_src_aux = rcx;
+        Xbyak::Reg64 reg_dst_aux = rdi;
+        mov(addr_work_amount_bk, reg_work_amount);
+        mov(addr_oc_off_bk, reg_oc_off);
+
+        Xbyak::Label tail_label;
+        cmp(addr_vector_num, 0);
+        je(tail_label, T_NEAR);
+
+        // unroll_size = vec_num >= unroll_size ? unroll_size : vec_num;
+        mov(addr_unroll_size, 4);  // default is 4 for addr_unroll_size
+        mov(rax, addr_unroll_size);
+        Xbyak::Label label_reset_unroll_size_end;
+        cmp(rax, addr_vector_num);
+        jle(label_reset_unroll_size_end, T_NEAR);
+        mov(rax, addr_vector_num);
+        mov(addr_unroll_size, rax);
+        L(label_reset_unroll_size_end);
+
+        // last unroll_size: vector_num % unroll_size
+        Xbyak::Label label_reset_last_unroll_size;
+        Xbyak::Label label_reset_last_unroll_size_end;
+        mov(rax, addr_vector_num);
+        xor_(rdx, rdx);
+        div(addr_unroll_size);  // rdx
+        cmp(rdx, 0);
```

Comment:
As I understand it, `rdx = rax / addr_unroll_size = addr_vector_num / addr_unroll_size`. And code block here https://github.com/chenhu-wang/openvino/blob/chenhu/shape_agnostic_jit_kernel_for_MVN/src/plugins/intel_cpu/src/nodes/mvn.cpp#L1450-L1458 assures `addr_vector_num >= addr_unroll_size`. So it seems `rdx == 0` will never be true. Did I miss anything?

Reply:
rdx is the remainder(rax % addr_unroll_size), not rax / addr_unroll_size.